code/modules/reagents/chemistry/reagents/unique/eigenstasium.dm

```diff
@@ -0,0 +1,251 @@
+			addictCyc3 = 0 //counter
+			alt_clone = null
+	addictCyc3++
+	do_teleport(living_mob, get_turf(living_mob), 2, no_effects=TRUE) //Teleports player randomly
+	do_sparks(5, FALSE, living_mob)
+	..()
+
+/datum/reagent/eigenstate/addiction_act_stage4(mob/living/living_mob) //Thanks for riding Fermichem's wild ride. Mild jitter and player buggery.
+	if(alt_clone)//catch any stragilers
+		var/mob/living/carbon/clone = alt_clone
+		do_sparks(5,FALSE,clone)
+		qdel(clone) 
+		living_mob.visible_message("[living_mob] is snapped across to a different alternative reality!")
+		alt_clone = null
+	if(addiction_stage == 31)
+		SEND_SIGNAL(living_mob, COMSIG_ADD_MOOD_EVENT, "Eigentrip", /datum/mood_event/eigentrip, creation_purity)
+		SSblackbox.record_feedback("tally", "chemical_reaction", 1, "Eigenstasium wild rides ridden")
+		do_sparks(5, FALSE, living_mob)
+		do_teleport(living_mob, get_turf(living_mob), 2, no_effects=TRUE) //teleports clone so it's hard to find the real one!
+		do_sparks(5, FALSE, living_mob)
+		living_mob.Sleeping(100)
+		living_mob.Jitter(50)
+		to_chat(living_mob, "<span class='warning'>You feel your eigenstate settle, snapping an alternative version of yourself into reality.</span>")
+		living_mob.emote("me",1,"flashes into reality suddenly, gasping as they gaze around in a bewildered and highly confused fashion!",TRUE)
+		log_game("FERMICHEM: [living_mob] ckey: [living_mob.key] has become an alternative universe version of themselves.")
+		//new you new stuff
+		living_mob.reagents.remove_all(1000)
+		var/datum/component/mood/mood = living_mob.GetComponent(/datum/component/mood)
+		mood.remove_temp_moods() //New you, new moods.
+		var/mob/living/carbon/human/human_mob = living_mob
+		if(!human_mob)
+			return
+		if(prob(1))//low chance of the alternative reality returning to monkey
+			var/obj/item/organ/tail/monkey/monkey_tail = new (human_mob.loc)
+			monkey_tail.Insert(human_mob)
+		human_mob.dna?.species?.randomize_main_appearance_element(human_mob)
+		human_mob.dna?.species?.randomize_active_underwear(human_mob)
+
+
+	if(prob(20))
+		do_sparks(5, FALSE, living_mob)
+	..()
+
+///Lets you link lockers together
+/datum/reagent/eigenstate/expose_turf(turf/exposed_turf, reac_volume)
+	. = ..()
+	if(creation_purity < 0.8)
+		return
+	var/obj/structure/closet/first
+	var/obj/structure/closet/previous
+	for(var/obj/structure/closet/closet in exposed_turf.contents)
+		if(closet.eigen_target)
+			closet.visible_message("[closet] fizzes, it's already linked to something else!")
+			continue
+		if(!previous)
+			first = closet
+			previous = closet
+			continue
+		closet.convert_to_eigenlocker(previous)
+		previous = closet
+	if(!first)
+		return
+	if(previous == first)
+		return
+	first.convert_to_eigenlocker(previous)
+	first.visible_message("The lockers' eigenstates spilt and merge, linking each of their contents together.")
+
+//eigenstate END
+
+//delete me
+/datum/reagent/test
+	name = "testeig"
```

Comment:
```suggestion
	name = "testing"
```

Reply:
Oops I forgot to delete that, It was just so I didn't have to sit through the addiction everytime